Models/Text/BytePairEncoding.swift

```diff
@@ -144,6 +211,29 @@ extension BytePairEncoder {
         return parts
     }
 
+    /// Uses the given regex to split a token into individual glossary terms.
+    ///
+    /// - Parameters:
+    ///   - token: Full text.
+    ///   - glossaryRegex: Regular expression for segmenting the given token.
+    /// - Returns: Array of substrings that match the given regex.
+    internal static func splitWithDelimitersGpt2(
```

Comment:
Two suggestions:
- The method name should form a noun phrase, since the method is non-side-effectful.
- Abbreviations like `GPT2` or [`JSON`](https://developer.apple.com/documentation/foundation/jsonencoder) should be fully capitalized, unless they appear as the first word in a camelcase word (e.g. `gpt2XXX` or `jsonEncoder`).
```suggestion
    internal static func splittingWithDelimitersGPT2(
```

Reply:
That's good clarification about acronym capitalization, thanks!! Since I combined this function with an enum, I renamed it to `splittingWithDelimiters`.